### ISSUE TITLE
Middleware runs more than once

### ISSUE THREAD

**Issue by rmvermeulen** [nestjs/nest#779]
## I'm submitting a...
<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Logger middleware called more than once per request

## Expected behavior
Middleware is called once for a request

## Minimal reproduction of the problem with instructions

Using the latest [cats sample](https://github.com/nestjs/nest/tree/master/sample/01-cats-app), with the `cats.module.ts` applying the logger middleware
```ts
// logger middleware
@Injectable()
export class LoggerMiddleware implements NestMiddleware {
  resolve(context: string): MiddlewareFunction {
    return (req, res, next) => {
      console.log(`[${context}] [${req.method}] [${req.baseUrl}] Request...`);
      next();
    };
  }
}
// cats module
export class CatsModule implements NestModule {
  configure(consumer: MiddlewareConsumer) {
    consumer.apply(LoggerMiddleware)
      .with('CatsModule')
      .forRoutes(CatsController);
  }
}
```
After `curl http://localhost:3000/cats/12` the server outputs
```
[CatsModule] [GET] [/cats] Request...
[CatsModule] [GET] [/cats/12] Request...
Before...
After... 0ms
```

Note that if I use `.forRoutes('/cats')` (instead of the controller class), or apply the middleware in main.ts with `app.use(...)`,  I get the expected behaviour.

## Environment

<pre><code>
Nest version: 5.0.1

 
For Tooling issues:
- Node version: 8.9.4  
- Platform: Mac 

Others:

</code></pre>

**Comment by cojack** [comment]
@kamilmysliwiec could you point issues in commits? It will be nice hooked together ;)

**Comment by kamilmysliwiec** [comment]
Fixed in the v5.1.0
